File: linked_list/linked_list/linked_list.cpp
#include "variables.h"
#include "stdafx.h"
#include "iostream"
#include "string"

using namespace std;

struct Node
{
	int data;
	struct Node *next;
};

class LinkedList {
private:
	Node *head;
	Node *tail;

	LinkedList() {
		head = null;
		tail = null;
	}

public:

	void insert(int element) {
		Node* node = new Node(malloc(sizeof(struct Node)));
		size++;
	}

	void display() {

	}

	void deleteElement(int position) {

	}

	int getSize() {
		return size;
	}

	int isFull() {
		return size == 10;
	}
};

int main() {
	int answer_number;
	int element;
	string yes_no_answer;
	LinkedList list = new LinkedList();
	do
	{
		cout << "1: insert" << endl;
		cout << "2: display" << endl;
		cout << "3: delete" << endl;
		cout << "4: isFull?" << endl;
		cout << "5: isEmpty" << endl;
		cout << "6: getSize" << endl;
		cout << "Answer: ";
		cin >> answer_number;
		switch (answer_number)
		{
		case 1:
			if (!list.isFull())
			{
				cout << "Element: ";
				list->insert(element);
			}
			else {
				cout << "List is full!" << endl;
			}
			break;
		case 2:
			break;
		case 3:
			break;
		case 4:
			break;
		case 5:
			break;
		case 6:
			break;
		default:
			throw new runtime_error("Incorrect answer!");
		}

		cout << "Continue? (y/n) ";
		cin >> yes_no_answer;
	} while (yes_no_answer == "y" || yes_no_answer == "Y");

	return 0;
}

File: LinkedList/LinkedList/LinkedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LinkedList
{
    interface IList
    {
        bool IsEmpty();

        bool IsFull();

        int GetSize();

        void InsertFirst(int element);

        void InsertAtPosition(int element, int position);

        void InsertLast(int element);

        void Display();

        void DeleteAtPosition(int position);

        int NegativeElementQuantity();
    }

    class Node
    {
        int Data;
        public Node Prev;
        public Node Next;

        public Node(int data)
        {
            Data = data;
        }
    }

    class LinkedList : IList
    {
        private int Size = 0;
        private int MaxSize;
        private Node Head = null;
        private Node Tail = null;

        public LinkedList(int maxSize)
        {
            MaxSize = maxSize;
        }

        public void DeleteAtPosition(int position)
        {
            throw new NotImplementedException();
        }

        public void Display()
        {
            throw new NotImplementedException();
        }

        public int GetSize()
        {
            return Size;
        }

        public void InsertAtPosition(int element, int position)
        {
            throw new NotImplementedException();
        }

        public void InsertFirst(int element)
        {
            Node node = new Node(element);
            if(IsEmpty())
            {
                Head = node;
                Tail = Head;
            } else
            {
                Head.Prev = node;
                node.Next = Head;
                Head = node;
            }
            Size++;
        }

        public void InsertLast(int element)
        {
            throw new NotImplementedException();
        }

        public bool IsEmpty()
        {
            return Head == null;
        }

        public bool IsFull()
        {
            return Size == MaxSize;
        }

        public int NegativeElementQuantity()
        {
            throw new NotImplementedException();
        }

        static void Main(string[] args)
        {
        }
    }
}
